source/editableText.py

```diff
@@ -42,21 +43,30 @@ class EditableText(ScriptableObject):
 	#: Whether or not to announce text found before the caret on a new line (e.g. auto numbering)
 	announceNewLineText=True
 
-	def _hasCaretMoved(self, bookmark, retryInterval=0.01, timeout=0.03):
+	def _hasCaretMoved(self, bookmark, retryInterval=0.01, timeout=None):
 		"""
 		Waits for the caret to move, for a timeout to elapse, or for a new focus event or script to be queued.
 		@param bookmark: a bookmark representing the position of the caret before  it was instructed to move
 		@type bookmark: bookmark
 		@param retryInterval: the interval of time in seconds this method should  wait before checking the caret each time.
 		@type retryInterval: float 
-		@param timeout: the over all amount of time in seconds the method should wait before giving up completely.
+		@param timeout: the over all amount of time in seconds the method should wait before giving up completely,
+			C{None} to use the value from the configuration.
 		@type timeout: float
 		@return: a tuple containing a boolean denoting whether this method timed out, and  a TextInfo representing the old or updated caret position or None if interupted by a script or focus event.
 		@rtype: tuple
- 		"""
+		"""
+		if timeout is None:
+			timeout = config.conf["editableText"]["caretMoveTimeout"]
+		else:
+			# This function's arguments are in seconds, but we want ms.
+			timeout *= 1000

```

Comment:
Since timeout is documented on the function to be in seconds, I think it could be easy to miss that its converted into milliseconds. Perhaps store the milliseconds in a new var, `timeoutMs`